pydantic/main.py

```diff
@@ -810,6 +564,383 @@ def model_parametrized_name(cls, params: tuple[type[Any], ...]) -> str:
+            raise PydanticUserError('You must set the config attribute `from_attributes=True` to use from_orm')
+        return cls.model_validate(obj)
+
+    @classmethod
+    def construct(cls: type[Model], _fields_set: set[str] | None = None, **values: Any) -> Model:
+        warnings.warn('The `construct` method is deprecated; use `model_construct` instead.', DeprecationWarning)
+        return cls.model_construct(_fields_set=_fields_set, **values)
+
+    def copy(
+        self: Model,
+        *,
+        include: AbstractSetIntStr | MappingIntStrAny | None = None,
+        exclude: AbstractSetIntStr | MappingIntStrAny | None = None,
+        update: typing.Dict[str, Any] | None = None,  # noqa UP006
+        deep: bool = False,
+    ) -> Model:
+        """
+        This method is now deprecated; use `model_copy` instead. If you need include / exclude, use:
+
+            data = self.model_dump(include=include, exclude=exclude, round_trip=True)
+            data = {**data, **(update or {})}
+            copied = self.model_validate(data)
+        """
+        warnings.warn(
+            'The `copy` method is deprecated; use `model_copy` instead. '
+            'See the docstring of `BaseModel.copy` for details about how to handle `include` and `exclude`.',
+            DeprecationWarning,
+        )
+
+        values = dict(
+            self._iter(to_dict=False, by_alias=False, include=include, exclude=exclude, exclude_unset=False),
+            **(update or {}),
+        )
+
+        # new `__fields_set__` can have unset optional fields with a set value in `update` kwarg
+        if update:
+            fields_set = self.__fields_set__ | update.keys()
+        else:
+            fields_set = set(self.__fields_set__)
+
+        # removing excluded fields from `__fields_set__`
+        if exclude:
+            fields_set -= set(exclude)
+
+        return self._copy_and_set_values(values, fields_set, deep=deep)
+
+    @classmethod
+    def schema(
+        cls, by_alias: bool = True, ref_template: str = DEFAULT_REF_TEMPLATE
+    ) -> typing.Dict[str, Any]:  # noqa UP006
+        warnings.warn('The `schema` method is deprecated; use `model_json_schema` instead.', DeprecationWarning)
+        return cls.model_json_schema(by_alias=by_alias, ref_template=ref_template)
+
+    @classmethod
+    def schema_json(
+        cls, *, by_alias: bool = True, ref_template: str = DEFAULT_REF_TEMPLATE, **dumps_kwargs: Any
+    ) -> str:
+        warnings.warn(
+            'The `schema_json` method is deprecated; use `model_json_schema` and json.dumps instead.',
+            DeprecationWarning,
+        )
+        from .json import pydantic_encoder
+
+        return cls.model_config['json_dumps'](
+            cls.model_json_schema(by_alias=by_alias, ref_template=ref_template),
+            default=pydantic_encoder,
+            **dumps_kwargs,
+        )
+
+    @classmethod
+    def validate(cls: type[Model], value: Any) -> Model:
```

Comment:
This was internal and wasn't documented, not sure we need it anymore.

Reply:
My I think we just raise an error, there's no obvious replacement for it.

Reply:
Looking at the definition from v1, it seems `model_validate` does the same thing for the most part: https://github.com/pydantic/pydantic/blob/11f7f9019016659e81c96481ffe5c42637489cf3/pydantic/main.py#L710-L719 

I realize there is the logic at the top related to what to do with subclass instances but that too seems handled by `model_validate` now with the recent changes you made in pydantic core. Is it really different enough to be worth removing instead of just adding a deprecation warning? I could add a note in the warning that the behavior may now be slightly different.

If you really think it's better to remove it, I'd suggest raising an error that says to replace it with model_validate, but I'm not sure what type of error to raise

Reply:
Ok, let's go with a warning.